modules/core/src/main/java/org/apache/fluo/core/oracle/OracleServer.java

```diff
@@ -410,15 +417,18 @@ public void takeLeadership(CuratorFramework curatorFramework) throws Exception {
       }
 
       synchronized (this) {
+        // FLUO-1000
+        isLeader = true;
```

Comment:
Was this moved to avoid race condition with `updateAdvertisedGcTimestamp()` and/or `isLeader()`?

Reply:
Yes

Reply:
@kpm1985  after 9b5277c9c79bb3a52bb879af591d1c51ae9476cc, I think this may be the only change in this PR that still need to be made.  It still makes sense to set isLeader before starting gcTsTracker.